File: t/CMDTEST_exit_zero.rb

require "selftest_utils"

class CMDTEST_exit_zero < Cmdtest::Testcase

  include SelftestUtils

  def test_exit_zero_CORRECT
    create_CMDTEST_foo [
      "cmd 'true.rb' do",
      "    exit_zero",
      "end",
    ]

    cmd_cmdtest do
      stdout_equal [
        "### true.rb",
      ]
    end
  end

  #----------------------------------------

  def test_exit_zero_INCORRECT
    create_CMDTEST_foo [
      "cmd 'false.rb' do",
      "    exit_zero",
      "end",
    ]

    cmd_cmdtest do
      stdout_equal [
        "### false.rb",
        "--- ERROR: expected zero exit status, got 1",
      ]
      exit_nonzero
    end
  end

  #----------------------------------------

  def test_exit_zero_INCORRECT_18
    create_CMDTEST_foo [
      "cmd 'exit.rb 18' do",
      "    exit_zero",
      "end",
    ]

    cmd_cmdtest do
      stdout_equal [
        "### exit.rb 18",
        "--- ERROR: expected zero exit status, got 18",
      ]
      exit_nonzero
    end
  end

end

File: python/t/CMDTEST_exit_zero.py

from selftest_utils import SelftestUtils

class TC_exit_zero(SelftestUtils, TestCase):

    def test_exit_zero_CORRECT(self):
        self.create_CMDTEST_foo(
            'with self.cmd("true") as c:',
            '    c.exit_zero()',
        )

        with self.cmd_cmdtest() as c:
            c.stdout_equal([
                "### cmdline: true",
            ])

    def test_exit_zero_INCORRECT(self):
        self.create_CMDTEST_foo(
            'with self.cmd("false") as c:',
            '    c.exit_zero()',
        )

        with self.cmd_cmdtest() as c:
            c.stdout_equal([
                "### cmdline: false",
                "--- ERROR: exit_zero",
                "actual: 1",
                "expect: 0",
                "",
            ])
            c.exit_nonzero()

    def test_exit_zero_INCORRECT_18(self):
        self.create_CMDTEST_foo(
            'with self.cmd("exit 18") as c:',
            '    c.exit_zero()',
        )

        with self.cmd_cmdtest() as c:
            c.stdout_equal([
                "### cmdline: exit 18",
                "--- ERROR: exit_zero",
                "actual: 18",
                "expect: 0",
                "",
            ])
            c.exit_nonzero()
